src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MemberLookup.cs

```diff
@@ -106,7 +106,7 @@ private void RecordType(AggregateType type, Symbol sym)
             // If it is first, record it.
             if (_swtFirst == null)
             {
-                _swtFirst.Set(sym, type);
+                _swtFirst.Set(sym, type); // NullReferenceException possible!
```

Comment:
```_swtFirst``` actually is never null since it is initialized only once in constructor.
The wierd thing here is that ```SymWithType``` overloads the equality operator so that it equals to null when one of its field is null.